File: src/main/java/io/github/oxnz/Ingrid/dts/TxRequest.java
package io.github.oxnz.Ingrid.dts;

public class TxRequest {
    final TxCategory cat;
    final String ref;
    final String state;
    final String city;

    public TxRequest(String ref, TxCategory cat, String state, String city) {
        this.ref = ref;
        this.cat = cat;
        this.state = state;
        this.city = city;
    }
}

File: src/main/scala/io/github/oxnz/Ingrid/tx/TxRequest.scala
package io.github.oxnz.Ingrid.tx

class TxRequest(val ref: String, val cat: TxCategory, val state: String, val city: String) {
  require(ref != null, "ref is required")
  require(cat != null, "cat is required")
  require(state != null, "state is required")
  require(city != null, "city is required")
}
